src/Protocol/AbstractProtocol.php

```diff
@@ -206,7 +206,14 @@ protected function _connect($remote)
         $errorStr = '';
 
         // open connection
-        $this->socket = @stream_socket_client($remote, $errorNum, $errorStr, self::TIMEOUT_CONNECTION);
+        set_error_handler(
+            function ($error, $message = '') {
+                throw new Exception\RuntimeException(sprintf('Could not open socket: %s', $message), $error);
```

Comment:
isn't such code faulty that error handler is not restored when exception is thrown?

point being that throwing exception not necessarily terminate the program, it may resume working with this error handler being still set.